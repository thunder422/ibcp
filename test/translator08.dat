##################################################
#  Translator Test #8: More Errors (2010-06-29)  #
##################################################

Z = A$ + B$ + C$
Z = MID$(A*B+C,1)
Z$ = MID$(A*B+C,1)
Z$ = MID$(Y$,A$ + B$ + C$)
Z = A <> B$
Z = A <> B$ < C$
Z = A + B$ + C$
Z = A < B$ + C$
Z = A + B$ + C%
Z = A < B$ + C%
Z = A <> B$ + C$ < D$
Z = A + B$ + C$ < D$
Z = A + (B$ + C$ < D$)
Z$ = A$ + B$ + C%
Z$ = A$ + B * C
A$,RIGHT$(B,2) = C$
A$=RIGHT$(B,2) = C$
Z=MID$(A$,2) < "B"
Z=MID$(A$,2) = B$
Z(INT(A+B))=C
Z(A$)=C
Z(A)=Y(A$)

# added more array assignments, non-errors (2011-02-01)
Z(A),Y(A)=X(A)
Z(A,Y(B))=X(A)
Z%(A%),Y%(A%)=X%(A%)
Z$(A%),Y$(A%)=X$(A%)

# added more command error tests (2011-03-10)
TAB(10)=A
+
NOT
A PRINT B
A(I) PRINT B
MID$(A$ PRINT,4)=""
LET (
A,PRINT
A = (0 LET 5)
A = 0 PRINT 5
A = (0 PRINT 5)
A$ = MID$(B$ PRINT, 4)
A = INT(0 LET 5)
A = INT(0 PRINT 5)
A$ = MID$(B$, 4 PRINT)
A$ = MID$(B$, 4, 5 PRINT)
MID$((A$),4)=B$
MID$(-A$,4)=B$
MID$(+A$,4)=B$
FNA=B
FNA(I)=B
LET FNA=B
LET FNA(I)=B
C,FNA=B
C,FNA(I)=B
LET C,FNA=B
LET C,FNA(I)=B
C,FNA,D=B
C,FNA(I),D=B
LET C,FNA,D=B
LET C,FNA(I),D=B
