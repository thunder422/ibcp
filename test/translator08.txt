
Input: Z = A$ + B$ + C$
           ^^^^^^^^^^^^-- expected numeric expression

Input: Z = MID$(A*B+C,1)
                ^^^^^-- expected string expression

Input: Z$ = MID$(A*B+C,1)
                 ^^^^^-- expected string expression

Input: Z$ = MID$(Y$,A$ + B$ + C$)
                    ^^^^^^^^^^^^-- expected numeric expression

Input: Z = A <> B$
                ^^-- expected numeric expression

Input: Z = A <> B$ < C$
Output: Z<ref> A B$ C$ <$ <>%2 CvtDbl Assign 

Input: Z = A + B$ + C$
               ^^-- expected numeric expression

Input: Z = A < B$ + C$
               ^^^^^^^-- expected numeric expression

Input: Z = A + B$ + C%
               ^^-- expected numeric expression

Input: Z = A < B$ + C%
                    ^^-- expected string expression

Input: Z = A <> B$ + C$ < D$
Output: Z<ref> A B$ C$ +$ D$ <$ <>%2 CvtDbl Assign 

Input: Z = A + B$ + C$ < D$
               ^^-- expected numeric expression

Input: Z = A + (B$ + C$ < D$)
Output: Z<ref> A B$ C$ +$ D$ <$ +%2 Assign 

Input: Z$ = A$ + B$ + C%
                      ^^-- expected string expression

Input: Z$ = A$ + B * C
                 ^^^^^-- expected string expression

Input: A$,RIGHT$(B,2) = C$
                 ^-- expected string variable

Input: A$=RIGHT$(B,2) = C$
                 ^-- expected string expression

Input: Z=MID$(A$,2) < "B"
Output: Z<ref> A$ 2% MID2$( "B" <$ CvtDbl Assign 

Input: Z=MID$(A$,2) = B$
Output: Z<ref> A$ 2% MID2$( B$ =$ CvtDbl Assign 

Input: Z(INT(A+B))=C
Output: A B + INT( CvtInt Z(<ref> C Assign 

Input: Z(A$)=C
         ^^-- expected numeric expression

Input: Z(A)=Y(A$)
              ^^-- expected numeric expression

Input: Z(A),Y(A)=X(A)
Output: A CvtInt Z(<ref> A CvtInt Y(<ref> A CvtInt X( AssignList 

Input: Z(A,Y(B))=X(A)
Output: A CvtInt B CvtInt Y( CvtInt Z(<ref> A CvtInt X( Assign 

Input: Z%(A%),Y%(A%)=X%(A%)
Output: A% Z%(<ref> A% Y%(<ref> A% X%( AssignList% 

Input: Z$(A%),Y$(A%)=X$(A%)
Output: A% Z$(<ref> A% Y$(<ref> A% X$( AssignList$ 

Input: TAB(10)=A
       ^^^^-- expected command

Input: +
       ^-- expected command

Input: NOT
       ^^^-- expected command

Input: A PRINT B
         ^^^^^-- expected equal or comma for assignment

Input: A(I) PRINT B
            ^^^^^-- expected equal or comma for assignment

Input: MID$(A$ PRINT,4)=""
               ^^^^^-- expected comma

Input: LET (
           ^-- expected item for assignment

Input: A,PRINT
         ^^^^^-- expected double variable

Input: A = (0 LET 5)
              ^^^-- expected operator or closing parentheses

Input: A = 0 PRINT 5
             ^^^^^-- expected operator or end-of-statement

Input: A = (0 PRINT 5)
              ^^^^^-- expected operator or closing parentheses

Input: A$ = MID$(B$ PRINT, 4)
                    ^^^^^-- expected operator or comma

Input: A = INT(0 LET 5)
                 ^^^-- expected operator or closing parentheses

Input: A = INT(0 PRINT 5)
                 ^^^^^-- expected operator or closing parentheses

Input: A$ = MID$(B$, 4 PRINT)
                       ^^^^^-- expected operator, comma or closing parentheses

Input: A$ = MID$(B$, 4, 5 PRINT)
                          ^^^^^-- expected operator or closing parentheses

Input: MID$((A$),4)=B$
            ^-- expected string variable

Input: MID$(-A$,4)=B$
            ^-- expected string variable

Input: MID$(+A$,4)=B$
            ^-- expected string variable

Input: FNA=B
Output: FNA<ref> B Assign 

Input: FNA(I)=B
          ^-- expected equal or comma for assignment

Input: LET FNA=B
Output: FNA<ref> B Assign'LET' 

Input: LET FNA(I)=B
              ^-- expected equal or comma for assignment

Input: C,FNA=B
Output: C<ref> FNA<ref> B AssignList 

Input: C,FNA(I)=B
            ^-- expected equal or comma for assignment

Input: LET C,FNA=B
Output: C<ref> FNA<ref> B AssignList'LET' 

Input: LET C,FNA(I)=B
                ^-- expected equal or comma for assignment

Input: C,FNA,D=B
Output: C<ref> FNA<ref> D<ref> B AssignList 

Input: C,FNA(I),D=B
            ^-- expected equal or comma for assignment

Input: LET C,FNA,D=B
Output: C<ref> FNA<ref> D<ref> B AssignList'LET' 

Input: LET C,FNA(I),D=B
                ^-- expected equal or comma for assignment

