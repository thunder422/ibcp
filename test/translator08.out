
Input: Z = A$ + B$ + C$
           ^^^^^^^^^^^^-- expected numeric expression

Input: Z = MID$(A*B+C,1)
                ^^^^^-- expected string expression

Input: Z$ = MID$(A*B+C,1)
                 ^^^^^-- expected string expression

Input: Z$ = MID$(Y$,A$ + B$ + C$)
                    ^^^^^^^^^^^^-- expected numeric expression

Input: Z = A <> B$
                ^^-- expected numeric expression

Input: Z = A <> B$ < C$
Output: Z = A <> B$ < C$ 

Input: Z = A + B$ + C$
               ^^-- expected numeric expression

Input: Z = A < B$ + C$
               ^^^^^^^-- expected numeric expression

Input: Z = A + B$ + C%
               ^^-- expected numeric expression

Input: Z = A < B$ + C%
                    ^^-- expected string expression

Input: Z = A <> B$ + C$ < D$
Output: Z = A <> B$ + C$ < D$ 

Input: Z = A + B$ + C$ < D$
               ^^-- expected numeric expression

Input: Z = A + (B$ + C$ < D$)
Output: Z = A + (B$ + C$ < D$) 

Input: Z$ = A$ + B$ + C%
                      ^^-- expected string expression

Input: Z$ = A$ + B * C
                 ^^^^^-- expected string expression

Input: A$,RIGHT$(B,2) = C$
                 ^-- expected string variable

Input: A$=RIGHT$(B,2) = C$
                 ^-- expected string expression

Input: Z=MID$(A$,2) < "B"
Output: Z = MID$(A$, 2) < "B" 

Input: Z=MID$(A$,2) = B$
Output: Z = MID$(A$, 2) = B$ 

Input: Z(INT(A+B))=C
Output: Z(INT(A + B)) = C 

Input: Z(A$)=C
         ^^-- expected numeric expression

Input: Z(A)=Y(A$)
              ^^-- expected numeric expression

Input: Z(A),Y(A)=X(A)
Output: Z(A), Y(A) = X(A) 

Input: Z(A,Y(B))=X(A)
Output: Z(A, Y(B)) = X(A) 

Input: Z%(A%),Y%(A%)=X%(A%)
Output: Z%(A%), Y%(A%) = X%(A%) 

Input: Z$(A%),Y$(A%)=X$(A%)
Output: Z$(A%), Y$(A%) = X$(A%) 

Input: TAB(10)=A
       ^^^^-- expected command

Input: +
       ^-- expected command

Input: NOT
       ^^^-- expected command

Input: A PRINT B
         ^^^^^-- expected equal or comma for assignment

Input: A(I) PRINT B
            ^^^^^-- expected equal or comma for assignment

Input: MID$(A$ PRINT,4)=""
               ^^^^^-- expected comma

Input: LET (
           ^-- expected item for assignment

Input: A,PRINT
         ^^^^^-- expected double variable

Input: A = (0 LET 5)
              ^^^-- expected operator or closing parentheses

Input: A = 0 PRINT 5
             ^^^^^-- expected operator or end-of-statement

Input: A = (0 PRINT 5)
              ^^^^^-- expected operator or closing parentheses

Input: A$ = MID$(B$ PRINT, 4)
                    ^^^^^-- expected operator or comma

Input: A = INT(0 LET 5)
                 ^^^-- expected operator or closing parentheses

Input: A = INT(0 PRINT 5)
                 ^^^^^-- expected operator or closing parentheses

Input: A$ = MID$(B$, 4 PRINT)
                       ^^^^^-- expected operator, comma or closing parentheses

Input: A$ = MID$(B$, 4, 5 PRINT)
                          ^^^^^-- expected operator or closing parentheses

Input: MID$((A$),4)=B$
            ^-- expected string variable

Input: MID$(-A$,4)=B$
            ^-- expected string variable

Input: MID$(+A$,4)=B$
            ^-- expected string variable

Input: FNA=B
Output: FNA = B 

Input: FNA(I)=B
          ^-- expected equal or comma for assignment

Input: LET FNA=B
Output: LET FNA = B 

Input: LET FNA(I)=B
              ^-- expected equal or comma for assignment

Input: C,FNA=B
Output: C, FNA = B 

Input: C,FNA(I)=B
            ^-- expected equal or comma for assignment

Input: LET C,FNA=B
Output: LET C, FNA = B 

Input: LET C,FNA(I)=B
                ^-- expected equal or comma for assignment

Input: C,FNA,D=B
Output: C, FNA, D = B 

Input: C,FNA(I),D=B
            ^-- expected equal or comma for assignment

Input: LET C,FNA,D=B
Output: LET C, FNA, D = B 

Input: LET C,FNA(I),D=B
                ^-- expected equal or comma for assignment

